File: code/backend/java/core/src/main/java/games/paths/core/model/match/MatchStatuses.java
package games.paths.core.model.match;

import java.util.List;
import java.util.Set;

/**
 * MatchStatuses - the lifecycle statuses of a {@code gaming_match} row.
 *
 * <p>A match is "stopped" (terminal) when it is {@code ENDED} or
 * {@code GAMEOVER}; only stopped matches may be deleted by an admin.</p>
 */
public final class MatchStatuses {

    public static final String CREATED  = "CREATED";
    public static final String RUNNING  = "RUNNING";
    public static final String PAUSED   = "PAUSED";
    public static final String ENDED    = "ENDED";
    public static final String GAMEOVER = "GAMEOVER";

    /** Every valid status, in lifecycle order. */
    public static final List<String> ALL = List.of(CREATED, RUNNING, PAUSED, ENDED, GAMEOVER);

    /** Terminal statuses — a match in one of these is "stopped" and deletable. */
    public static final Set<String> TERMINAL = Set.of(ENDED, GAMEOVER);

    private MatchStatuses() {
    }

    /** Returns true when {@code status} is one of the valid match statuses. */
    public static boolean isValid(String status) {
        return status != null && ALL.contains(status);
    }

    /** Returns true when {@code status} is a terminal (stopped) status. */
    public static boolean isTerminal(String status) {
        return status != null && TERMINAL.contains(status);
    }
}

File: code/backend/php/src/Core/Domain/Matches/MatchStatuses.php
<?php

namespace Games\Paths\Core\Domain\Matches;

/**
 * MatchStatuses - the lifecycle statuses of a gaming_match row.
 *
 * A match is "stopped" (terminal) when it is ENDED or GAMEOVER; only stopped
 * matches may be deleted by an admin.
 */
final class MatchStatuses
{
    public const CREATED  = 'CREATED';
    public const RUNNING  = 'RUNNING';
    public const PAUSED   = 'PAUSED';
    public const ENDED    = 'ENDED';
    public const GAMEOVER = 'GAMEOVER';

    /** Every valid status, in lifecycle order. */
    public const ALL = [self::CREATED, self::RUNNING, self::PAUSED, self::ENDED, self::GAMEOVER];

    /** Terminal statuses — a match in one of these is "stopped" and deletable. */
    public const TERMINAL = [self::ENDED, self::GAMEOVER];

    private function __construct()
    {
    }

    public static function isValid(?string $status): bool
    {
        return $status !== null && in_array($status, self::ALL, true);
    }

    public static function isTerminal(?string $status): bool
    {
        return $status !== null && in_array($status, self::TERMINAL, true);
    }
}
